File: java/selenium-cucumber-testng/src/test/java/com/browserstack/pages/DropdownPage.java
package com.browserstack.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.ui.Select;

public class DropdownPage extends BasePage {

    @FindBy(id = "dropdown")
    private WebElement dropdown;

    public DropdownPage(WebDriver driver) {
        super(driver);
    }

    public void selectOptionByValue(String value) {
        Select select = new Select(dropdown);
        select.selectByValue(value);
    }

    public void selectOptionByVisibleText(String text) {
        Select select = new Select(dropdown);
        select.selectByVisibleText(text);
    }

    public String getSelectedOption() {
        Select select = new Select(dropdown);
        return select.getFirstSelectedOption().getText();
    }
}

File: python/pip/common_selenium/dropdown_page.py
from .base_page import BasePage
from selenium.webdriver.remote.webdriver import WebDriver
from selenium.webdriver.common.by import *
from selenium.webdriver.support.select import Select
from typing import Callable, Tuple, Any


class DropdownPage(BasePage):
    dropdown: Tuple[str, str]

    def __init__(self, driver: WebDriver):
        super().__init__(driver)
        self.dropdown = (By.ID, "dropdown")

    def run_test(self, assert_provider: Callable[..., Any]):
        self.select_option_by_value(self.dropdown, "1")
        assert_provider(self.get_selected_option(self.dropdown) == "Option 1")

        self.select_option_by_value(self.dropdown, "2")
        assert_provider(self.get_selected_option(self.dropdown) == "Option 2")

    def select_option_by_value(self, selector_select, value):
        select = self.get_clickable_element(selector_select)
        Select(select).select_by_value(value=value)

    def select_option_by_visible_test(self, selector_select, value):
        select = self.get_clickable_element(selector_select)
        Select(select).select_by_value(value=value)

    def get_selected_option(self, selector_select):
        select = self.get_clickable_element(selector_select)
        return Select(select).first_selected_option.text
